### PR TITLE
Revise function call & Realm

### PR DESCRIPTION
**Pull request by ksh8281**
* Revise Object::getPrototypeFromConstructor. we should get prototype from context of consturctor
* We should pass Optional<Object*> instead of Value with newTarget.
* Give correct prototype value for function prototypes and functions
* Implement %ThrowTypeError% correctly

Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

File: src/interpreter/ByteCodeInterpreter.cpp [hunk 1]
```diff
@@ -2149,20 +2149,21 @@ NEVER_INLINE void ByteCodeInterpreter::createFunctionOperation(ExecutionState& s
 
     bool isGenerator = cb->isGenerator();
     bool isAsync = cb->isAsync();
-    Object* proto = state.context()->globalObject()->functionPrototype();
+    Context* functionRealm = code->m_codeBlock->context();
```
Comment (clover2123) [inline src/interpreter/ByteCodeInterpreter.cpp:2152]: Minor suggestion.
`code->m_codeBlock` could be replaced by `cb`

File: src/runtime/GlobalObjectBuiltinError.cpp [hunk 2]
```diff
@@ -134,7 +138,15 @@ void GlobalObject::installError(ExecutionState& state)
 
     // http://www.ecma-international.org/ecma-262/5.1/#sec-13.2.3
     // 13.2.3 The [[ThrowTypeError]] Function Object
-    m_throwTypeError = new NativeFunctionObject(state, NativeFunctionInfo(state.context()->staticStrings().ThrowTypeError, builtinErrorThrowTypeError, 0, NativeFunctionInfo::Strict));
+    m_throwTypeError = new NativeFunctionObject(state, NativeFunctionInfo(AtomicString(), builtinErrorThrowTypeError, 0, NativeFunctionInfo::Strict));
+    m_throwTypeError->markThisObjectDontNeedStructureTransitionTable();
```
Comment (clover2123) [inline src/runtime/GlobalObjectBuiltinError.cpp:142]: `setGlobalIntrinsicObject` could replace `markThisObjectDontNeedStructureTransitionTable` at here.
`setGlobalIntrinsicObject` is used for initialization of each intrinsic object.

File: src/interpreter/ByteCodeInterpreter.cpp (hunk 1 above)
Comment (ksh8281) [reply src/interpreter/ByteCodeInterpreter.cpp]: I changed it! thanks

File: src/runtime/GlobalObjectBuiltinError.cpp (hunk 2 above)
Comment (ksh8281) [reply src/runtime/GlobalObjectBuiltinError.cpp]: I changed it! thanks

**Pull request merged**